### PR TITLE
Adding references to ref.bib

### PR DESCRIPTION
**Pull request by nnpereira**
## Description

Addresses subtasks mentioned in #1069 :

- [ ]  further_resources.rst
- [ ] plot_new_metrics.py
- [ ]  _tradeoff_curve_utilities.py (convex hull algorithm)

## Tests

- [X] no new tests required

### CODE REVIEW

**Comment by nnpereira** [comment]
@TamaraAtanasoska can you please review this PR ? 🙏

File: docs/refs.bib
Comment (riedgar-ms) [inline docs/refs.bib:865]: Should this be merged with `@inproceedings{buolamwini2018gender` ?

File: docs/refs.bib
Comment (riedgar-ms) [inline docs/refs.bib:858]: `@book{barocas2019fairness` sort of looks to be the same thing

**Review by alliesaizan — COMMENTED** [review]
Hi there, thank you for opening up this PR! Just wanted to verify that you looked through every page in the User Guide and added a missing reference to refs.bib where necessary. I did a quick scan and the only issue I see is small, that the section Performing a Fairness Assessment is missing a References section though a footnote is present in the guide. I'd welcome another set of eyes to review this PR as well to make sure I did not miss anything, so pinging one of the @fairlearn/fairlearn-maintainers to chime in.

File: docs/refs.bib
Comment (nnpereira) [reply docs/refs.bib:865]: Ah, I wondered about that. Since the reference I added is referring directly to the project website, how do you  suggest merging the two?

File: docs/refs.bib
Comment (nnpereira) [reply docs/refs.bib:858]: Agreed, I'll sub that reference instead.

Comment (nnpereira) [reply docs/refs.bib:858]: The book reference is mentioned earlier in that [ page](https://fairlearn.org/main/user_guide/further_resources.html), this reference is nested under the "Online demos, talks, tutorials" header.

**Comment by alliesaizan** [comment]
> hey @alliesaizan , this PR currently only addresses the subtasks I listed in the Description that were originallly flagged [here](https://github.com/fairlearn/fairlearn/issues/1069#issuecomment-1435231610).
> 
> However, I'm happy to do a pass through the User Guide and address the section you mentioned and update.

Ok, gotcha, did not review Roman's comment that the scope had been narrowed down to a few sections. If it's a quick change, I think it makes sense to address the missing footnote here rather than opening up a new PR for that small change.

File: docs/refs.bib
Comment (alliesaizan) [reply docs/refs.bib:865]: Is this reference only cited in the plot_new_metrics file? If so, I think it'd make sense to remove this from refs.bib and cite the research paper in the prose instead. Let me know your thoughts.

Comment (riedgar-ms) [reply docs/refs.bib:865]: Can you add a URL field to the BibTeX entry?

**Comment by TamaraAtanasoska** [comment]
@nnpereira yay for your first PR being almost done! as far as I understand from the above there are two more things to be addressed: 
- [creating a URL entry](https://github.com/fairlearn/fairlearn/pull/1466#discussion_r1885854287)
- [adding a references section in a file where it's missing](https://github.com/fairlearn/fairlearn/pull/1466#pullrequestreview-2500823236)

otherwise LGTM :) sometimes the doc build fails because it is unable to download some external datasets, if you see a red and a reason related to a resource being unavailable it can help to push an empty commit with: `git commit --allow-empty -m "Trigger Build"` to retrigger the build once more. it might not happen again, just FYI :)

File: docs/refs.bib
Comment (nnpereira) [reply docs/refs.bib:865]: @alliesaizan yep only referenced cited in`plot_new_metrics.py` and for sure @riedgar-ms i'll update the `@inproceedings{buolamwini2018gender}` with a url field.

**Comment by nnpereira** [comment]
thanks @TamaraAtanasoska  the build tip you mentioned was very handy!

just addressed the two things you mentioned, in addition to catching another link in `fairness_in_machine_learning.rst` that needed a citation swapping out a link to a talk to an existing reference.

**Review by TamaraAtanasoska — APPROVED** [review]
LGTM! Thanks for sticking with all the comments @nnpereira . Does this close the original issue? Or there are still some open?

**Comment by nnpereira** [comment]
It looks like the top two subtasks haven't been reviewed yet [here](https://github.com/fairlearn/fairlearn/issues/1069#issuecomment-1435231610) so I can look at those before we close!

**Pull request merged**
